ui/components/component-library/avatar-token/avatar-token.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { useState, useEffect, Ref } from 'react';
+import classnames from 'classnames';
+import { AvatarBase, AvatarBaseSize } from '../avatar-base';
+import {
+  DISPLAY,
+  AlignItems,
+  JustifyContent,
+  TextColor,
+  BorderColor,
+  BackgroundColor,
+} from '../../../helpers/constants/design-system';
+import { AvatarTokenProps, AvatarTokenSize } from './avatar-token.types';
+
+export const AvatarToken = React.forwardRef(
+  (
+    {
+      size = AvatarTokenSize.Md,
+      name,
+      src,
+      showHalo,
+      color = TextColor.textDefault,
+      backgroundColor = BackgroundColor.backgroundAlternative,
+      borderColor = BorderColor.transparent,
+      className = '',
+      ...props
+    }: AvatarTokenProps,
+    ref: Ref<HTMLElement>,
+  ) => {
+    const [showFallback, setShowFallback] = useState(false);
+
+    useEffect(() => {
+      setShowFallback(!src);
+    }, [src]);
+
+    const handleOnError = () => {
+      setShowFallback(true);
+    };
+
+    const fallbackString = name?.[0] ?? '?';
+
+    return (
+      <AvatarBase
+        ref={ref}
+        size={size as unknown as AvatarBaseSize}
```

Comment:
Change back to size and add `AvatarTokenSize` as a union type to `AvatarBase`
```suggestion
        size={size}
```

Reply:
Agreed, let's do that.